pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -2466,6 +2467,62 @@ func (r *ReconcileClusterDeployment) discoverAzureResourceGroup(cd *hivev1.Clust
 	return true
 }
 
+// discoverGCPNetworkProjectID is intended to retrofit GCP clusters which were a) adopted, or b) created before
+// support for shared VPC was added. It attempts to discover the network project ID for the cluster by parsing
+// the install-config.
+// No op if platform is not GCP.
+// No op if the field is already set (we don't validate that it is correct).
+// This is best effort. If things go wrong, we log but do not fail out. THIS WILL BLOCK DEPROVISIONING. If
+// this happens, the field must be set manually.
+// The return indicates whether the field was modified. If `true`, the caller is responsible for pushing
+// an Update() back to the server.
+func (r *ReconcileClusterDeployment) discoverGCPNetworkProjectID(cd *hivev1.ClusterDeployment, log log.FieldLogger) bool {
```

Comment:
Where is this function used?

Reply:
Did I forget the bit where it gets called? :sheepish:

Reply:
Fixed.